contracts/MultiSigWallet.sol

```diff
@@ -114,8 +104,7 @@ contract MultiSigWallet {
         validRequirement(_owners.length, _required)
     {
         for (uint i=0; i<_owners.length; i++) {
-            if (isOwner[_owners[i]] || _owners[i] == 0)
-                throw;
+            require(!isOwner[_owners[i]] && _owners[i] > 0);
```

Comment:
i think you've done a pretty good job inverting these statements - HOWEVER - there are some very subtle changes in how you are changing the check logic.  

if you wanted to ensure the code was exactly the same, i would wrap all of these instead with `!(CONDITION)` instead of rewriting them.  eg, `require (!(isOwner[_owners[i]] || _owners[i] == 0)` is exactly the same as the left side.

Reply:
from my quick eyeballing, i do think you've captured these correctly, but i'm not 100% sure.  i know that i've had many bugs in the past from subtle differences in inverting boolean checks

Reply:
Ideally tests should cover this mistakes.

Do you want to switch to !(CONDITION) approach now?

Reply:
i feel a bit torn - i feel in some cases you are actually doing a better job than the original!